### PR TITLE
Add an option to set config

### PR DESCRIPTION
**Pull request by arajasek**
A solution for #3674 that seems to work.

### CODE REVIEW

**Review by magik6k — COMMENTED** [review]
It doesn't seem to actually be used, repo.Config() will still get you the one in lotus dir
```
func (fsr *fsLockedRepo) Config() (interface{}, error) {
	fsr.configLk.Lock()
	defer fsr.configLk.Unlock()

	return fsr.loadConfigFromDisk()
}

func (fsr *fsLockedRepo) loadConfigFromDisk() (interface{}, error) {
	return config.FromFile(fsr.join(fsConfig), defConfForType(fsr.repoType))
}
```

**Comment by arajasek** [comment]
@magik6k I'm actually overwriting the config in the lotus dir, so the provided one will be used (and persisted in the lotus dir)

**Comment by arajasek** [comment]
(willing to be told that's terrible, though, it was just the easiest to keysmash)

File: cmd/lotus/daemon.go [hunk 1]
```diff
@@ -184,6 +188,10 @@ var DaemonCmd = &cli.Command{
 			return xerrors.Errorf("repo init error: %w", err)
 		}
 
+		if cctx.String("config") != "" {
+			r.SetConfigPath(cctx.String("config"))
+		}
```
Comment (magik6k) [inline cmd/lotus/daemon.go:193]: Shouldn't this be above r.Init?

Comment (arajasek) [reply cmd/lotus/daemon.go:193]: I don't think it makes a difference? If above Init, we won't create a default config in the default path (if missing), which seems neither good nor bad?

**Review by magik6k — APPROVED** [review]

**Pull request merged**